### PR TITLE
Makefile: add api-docs to bundle rule 

### PR DESCRIPTION
**Pull request by frzifus**
**Description:** <Describe what has changed.>

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -192,7 +192,7 @@ lint: golangci-lint
 
 # Generate code
 .PHONY: generate
-generate: controller-gen
+generate: api-docs controller-gen
```
Comment (pavolloffay) [inline Makefile:195]: the `api-docs` traget was remvoed from generate to improve developer UX by minimizing local changes when developing the operator.  #2421

Comment (swiatekm-sumo) [reply Makefile:195]: Was that actually necessary? `api-docs` is a noop if you haven't made any changes impacting the docs, so it should be fine to have it here.

Comment (pavolloffay) [reply Makefile:195]: What does the generate update? Is it the deepcopy? 

I think the api-docs generate should be bundled with the target to generate CRD

Comment (swiatekm-sumo) [reply Makefile:195]: `generate` only generates the deepcopy code, yes.

Comment (swiatekm-sumo) [reply Makefile:195]: But I do agree that `api-docs` should be a part of `bundle` rather than `generate`.

**Review by swiatekm-sumo — APPROVED** [review]

**Pull request merged**